required/firstaid/latex2e-first-aid-for-external-files.dtx

```diff
@@ -618,6 +618,58 @@
 %    {Temporary fixes for the kernel \\
 %     (until the next patch-level release)}
 %
+% This fixes GitHub issue 591. It is only needed in Lua\TeX\ and replaces just one
+% instance of \cs{tex\_par:D} with the following version which removes
+% other nodes in the current list first.
+%    \begin{macrocode}
+\ExplSyntaxOn
+\sys_if_engine_luatex:T
+  {
+    \newluafunction \g__para_end_empty_par_id_int
+    \exp_args:Nx \everyjob {
+      \exp_not:V \everyjob
+      \exp_not:N \lua_now:n {
+        local~texnest, flush_list, par_token = tex.nest, node.flush_list, token.create'tex_par:D'~
```

Comment:
why exactly does that need to be implemented using \everyjob?

Reply:
No Lua state is dumped (except for Lua bytecode registers and even they only dump bytecode), so after loading the format all Lua functions are lost. Therefore all `\luadef`/`token.set_lua` based commands defined in the format have to be implemented with `\everyjob`.

Reply:
you live and learn (or relearn, I think I did know that at one point)